core/schema/modulesource.go

```diff
@@ -42,7 +42,16 @@ func (s *moduleSchema) moduleSource(ctx context.Context, query *core.Query, args
 	switch src.Kind {
 	case core.ModuleSourceKindLocal:
 		if filepath.IsAbs(parsed.modPath) {
-			return nil, fmt.Errorf("local module source root path is absolute: %s", parsed.modPath)
+			cwdStat, err := query.Buildkit.StatCallerHostPath(ctx, ".", true)
+			if err != nil {
+				return nil, fmt.Errorf("failed to stat caller's current working directory: %w", err)
+			}
+
+			relPath, err := filepath.Rel(cwdStat.Path, parsed.modPath)
+			if err != nil {
+				return nil, fmt.Errorf("failed to make path relative: %w", err)
+			}
+			parsed.modPath = relPath
```

Comment:
Hm, just a note, I think this will probably cause issues with windows paths.

Reply:
Thanks, you're totally right. I updated the code, with a `crossPlatformRelativePath` function  corresponding tests. I figured that we couldn't rely on the `filepath.ToSlash()` as the core/schema runs inside a linux container, or, the toslash implementation relies on the os to guess the separator to use